### PR TITLE
(#3481) - Split up the android tests

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

**Comment by daleharvey** [comment]
Woot, green :) (will remember to remove from expected fails)

**Comment by nickcolley** [comment]
This is awesome :D

**Comment by nolanlawson** [comment]
What is `invert` and how is it working? It kinda looks like `invert` isn't doing anything, and the Android tests are just passing twice in a row... Yet I do see in Travis that 174 passed in one and 909 in the other, so I'm puzzled.

**Comment by daleharvey** [comment]
Heh `INVERT` I found trying to debug why the regex I was using for grep wasnt working (https://github.com/mochajs/mocha/blob/master/mocha.js#L1641) It inverts the grep so GREP=test.replication.js will only do replication tests, GREP=test.replication.js INVERT=true will test everything but the replication ones

**Comment by nolanlawson** [comment]
Oh, awesome! That is an amazingly helpful trick. +1 then.

**Comment by daleharvey** [comment]
Yeh the nice thing about it is I was starting on setting up a `SUITE1` vs `SUITE2`, but its hard to guarantee that all the tests will run, with this we can just add to the `GREP=test.replication.js,test.compaction.js` if we need to split it out further and ever test will run

**Comment by daleharvey** [comment]
Cheers, merged in https://github.com/pouchdb/pouchdb/commit/47944e2c507fc6904a565432e553ebc42e738cc1

File: bin/test-browser.js [hunk 1]
```diff
@@ -49,6 +49,9 @@ if (client.runner === 'saucelabs') {
   qs.saucelabs = true;
 }
 if (process.env.GREP) {
```
Comment (fzzzy) [inline bin/test-browser.js]: Uhh, shouldn't this be INVERT?

Comment (nolanlawson) [inline bin/test-browser.js]: whoops, good catch!

Comment (daleharvey) [inline bin/test-browser.js]: heh copy and paste fail, luckily we only use INVERT when GREP is defined but good catch, thanks :)

**Pull request closed without merging**